components/Polls/Poll.js

```diff
@@ -21,16 +22,43 @@ export default class Poll extends Component {
   }
 
 
+  componentDidMount() {
+    this.getOptions();
+  }
+
+  componentWillUnmount() {
+    firebaseDB.ref().off('value');
```

Comment:
I'm not sure that this will work for your listener below. Can you check and confirm this is working? (Try going to poll, going to a different component, and then manually modifying firebase database console)